lib/ansible/plugins/connection/ssh.py

```diff
@@ -701,6 +731,53 @@ def _add_args(self, b_command: list[bytes], b_args: t.Iterable[bytes], explanati
         display.vvvvv(u'SSH: %s: (%s)' % (explanation, ')('.join(to_text(a) for a in b_args)), host=self.host)
         b_command += b_args
 
+    def _populate_agent(self) -> str:
+        """Adds configured private key identity to the SSH agent. Returns a path to a file containing the public key."""
+        if self._populated_agent:
+            return self._populated_agent
+
+        if (auth_sock := C.config.get_config_value('SSH_AGENT')) == 'none':
+            raise AnsibleError('Cannot utilize private_key with SSH_AGENT disabled')
+
+        key_data = self.get_option('private_key')
+        passphrase = self.get_option('private_key_passphrase')
+        private_key = load_private_key(
+            to_bytes(key_data),
+            to_bytes(passphrase) if passphrase else None
+        )
+        public_key = private_key.public_key()
+        pubkey_msg = PublicKeyMsg.from_public_key(public_key)
+        fingerprint = pubkey_msg.fingerprint()
+        with SshAgentClient(auth_sock) as client:
+            if public_key not in client:
+                display.vvv(f'SSH: SSH_AGENT adding {fingerprint} to agent', host=self.host)
+                client.add(
+                    private_key,
+                    '[added by ansible]',
```

Comment:
Do we have an identifier we could add to this for traceability?

Reply:
pid-uid-timestamp?

Reply:
What would be the use case? Isn't the fingerprint an identifier already?

Reply:
it identifies the key, and while this 'should not' happen, people share keys, specially if they are sudoing/su to the same user to execute things

Reply:
Exactly!

Reply:
So the use case is to have more info as to what/who/when added (first) the key into the agent. This is probably only useful for when `ssh_agent=$path_to_existing_agent`.

Added. Now it's e.g. `[added by ansible: PID=11813, UID=501, EUID=501, TIME=1742482534.45826]`

Reply:
it might be overkill, but stuff like this is always nice to have vs wish to have in middle of debugging issues